src/pages/videos/videoCard/VideoCard.jsx

```diff
@@ -1,15 +1,21 @@
 import React from "react";
 import { useData } from "../../../context/dataContext";
+import { useNavigate } from "react-router-dom";
 import "./video-card.css";
 
 function VideoCard({ video }) {
   const { state, dispatch } = useData();
   const { _id, title, category } = video;
   console.log("state ", state);
 
+  const navigate = useNavigate();
+
+  const openSingleVideo = () => {
+    navigate(`/video/${_id}`);
+  };
   return (
     <>
-      <div className="video-card">
+      <div className="video-card" id={_id} onClick={openSingleVideo}>
```

Comment:
You can give the <Link> tag here so that you won't need to call a function. It will redirect you to the required streaming page. Just a suggestion though...